rust/arrow/src/compute/kernels/sort.rs

```diff
@@ -1560,7 +1891,14 @@ mod tests {
             Some(2),
             Some(17),
         ])) as ArrayRef];
-        test_lex_sort_arrays(input, expected);
+        test_lex_sort_arrays(input.clone(), expected, None);
+
+        let expected = vec![Arc::new(PrimitiveArray::<Int64Type>::from(vec![
+            Some(-1),
+            Some(0),
+            Some(2),
+        ])) as ArrayRef];
+        test_lex_sort_arrays(input, expected, Some(3));
```

Comment:
here too I recommend using a limit that is less than the length of the array

Reply:
Here input size is 4, expected size is 3.